crypto/dilithium_test.go

```diff
@@ -0,0 +1,141 @@
+package crypto
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDilithiumSignVerify(t *testing.T) {
+	t.Run("Generate all possible keys per mode", func(tt *testing.T) {
```

Comment:
It's a good practice to name tests for each **behavior** that you want to test. See [this](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#naming-your-tests) for a more detailed explanation with examples.

As it stands, the intent of many of the tests is ambiguous and puts the onus of the reader.

Reply:
good callout - will update